### DISCUSSION TITLE
Freezing position while maintaining velocities

### DISCUSSION THREAD

**Discussion by merspieler** [JSBSim-Team/jsbsim#1209] [Q&A] [answered]
Hey, I'm using JSBSim with flightgear and have the following problem:
I want to make an aircraft start mid-air. The problem I have with this is that it takes some time till the systems are all initialized and set up correctly, in this time the aircraft just "falls" uncontrollably (started with some forward velocity, so it can glide a bit but still loosing about 3000ft).
So my idea was to freeze the movement of the aircraft for a moment (seen that on some professional A320 sim I've been to)
I found out I can set this `forces/hold-down=true` and the aircraft freezes on the spot as expected so the aircraft can be set up correctly.
Problem is when I unfreeze, it has 0 velocity and starts dropping straight down.
Is there a way to freeze the movement but maintain the velocity after unfreezing it? (bonus would be if the velocities exposed to the aircraft would also still show the velocities at the time of freezing the sim, so auto pilot etc would not get false information while the movement is frozen)

**Comment by bcoconni** [comment]
You may try freezing the integration scheme of the aircraft position and attitude. For that you would set the following properties to 0:
* `simulation/integrator/rate/translational`
* `simulation/integrator/rate/rotational`

Once you've completed the aircraft setup, you would restore the values of these properties to their initial values. Not sure this will give the expected result, but at least it should prevent the aircraft from moving while updating the velocities and accelerations.

**Reply by bcoconni** [reply, marked as answer]
Oops ! **EDIT** The properties names were wrong, they actually are:
* `simulation/integrator/position/translational`
* `simulation/integrator/position/rotational`

**Reply by merspieler** [reply]
looks promising, it almost stops, and releasing the movement also works fine.
Only issue I've found is that the aircraft starts moving west for some reason which is a bit strange to me but I think this would already work for me as I think the period I freeze the aircraft for might be short enough that I can a) ignore it or b) account for that in the startup location.
Thank you very much so far

**Comment by dershow** [comment]
Earth rotation?

On December 21, 2024 7:19:55 PM EST, merspieler ***@***.***> wrote:
>looks promising, it almost stops, and releasing the movement also works fine.
>Only issue I've found is that the aircraft starts moving west for some reason which is a bit strange to me but I think this would already work for me as I think the period I freeze the aircraft for might be short enough that I can a) ignore it or b) account for that in the startup location.
>Thank you very much so far
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/JSBSim-Team/jsbsim/discussions/1209#discussioncomment-11640152
>You are receiving this because you are subscribed to this thread.
>
>Message ID: ***@***.***>

--Adam

**Comment by Warning39** [comment]
What sorts of systems take a while to initialize? In particular do any of them adjust any of the inceptors (throttle, stick, rudder pedals)?

In general I'd suggest getting the aircraft trimmed at your selected position, altitude and speed. If the systems don't make any changes to the inceptors, or change the mass distribution etc. during initialization then the aircraft will still be in trim once they are initialized.

**Reply by merspieler** [reply]
It's stuff which is in big parts not in JSBSim, like flight management systems, etc
